src/editors/sharedComponents/InsertLinkModal/index.jsx

```diff
@@ -0,0 +1,220 @@
+  const [blocksList, setBlocksList] = useState(null);
+  const [, setInvalidUrlInput] = useState(false);
+  const [inputUrlValue, setInputUrlValue] = useState('');
+  const dispatch = useDispatch();
+  const { selectedBlocks } = useSelector(selectors.insertlinkState);
+
+  const handleSearchedBlocks = (isSearched) => {
+    setBlocksSearched(isSearched);
+  };
+
+  const handleSelectedBlock = (blockSelectedFromList) => {
+    setBlocksSelected(blockSelectedFromList);
+    setInputUrlValue('');
+  };
+
+  const handleCloseLink = () => {
+    setSearchField('');
+    setBlocksSelected(null);
+  };
+
+  /* istanbul ignore next */
+  const handleSave = () => {
+    const editor = editorRef.current;
+    const urlPath = blockSelected?.lmsWebUrl || inputUrlValue;
+    const blockId = blockSelected?.blockId;
+    if (editor && urlPath) {
+      const validateUrl = isValidURL(urlPath);
+
+      if (!validateUrl) {
+        setInvalidUrlInput(true);
+        return;
+      }
+
+      const selectedRange = editor.selection.getRng();
+      const selectedText = editor.selection.getContent({ format: 'text' });
+
+      const newLinkNode = editor.dom.create('a', {
+        href: urlPath,
+        'data-mce-href': urlPath,
+        'data-block-id': blockId,
+        target: '_blank',
+      });
+
+      newLinkNode.textContent = selectedText;
+
+      selectedRange.deleteContents();
+      selectedRange.insertNode(newLinkNode);
+      // Remove empty "a" tags after replacing URLs
+      const editorContent = editor.getContent();
+      const modifiedContent = editorContent.replace(/<a\b[^>]*><\/a>/gi, '');
+      editor.setContent(modifiedContent);
+
+      dispatch(actions.addBlock({ [blockId]: blockSelected }));
+    }
+
+    onClose();
+  };
+
+  useEffect(() => {
+    const getBlocksList = async () => {
+      try {
+        const blocksData = await getBlocksFromCourse(courseId);
+        const { blocks: blocksResponse, root: rootBlocksResponse } = blocksData;
+        const blockListFormatted = formatBlocks(
+          blocksResponse,
+          rootBlocksResponse,
+        );
+        setBlocksList(blockListFormatted);
+      } catch (error) {
+        logError(error);
+      }
+    };
+
+    getBlocksList();
+  }, []);
+
+  useEffect(() => {
+    /* istanbul ignore next */
+    const editor = editorRef.current;
+    if (editor) {
+      const selectedHTML = editor.selection.getContent({ format: 'html' });
+      const regexDataBlockId = /data-block-id="([^"]+)"/;
+      const regexHref = /href="([^"]+)"/;
+      const matchDataBlockId = selectedHTML.match(regexDataBlockId);
+      const matchHreUrl = selectedHTML.match(regexHref);
+
+      // Extracting the value from the match
+      const dataBlockId = matchDataBlockId ? matchDataBlockId[1] : null;
+      const hrefUrl = matchHreUrl ? matchHreUrl[1] : null;
+      const blockSelectedUrl = selectedBlocks?.[dataBlockId]?.lmsWebUrl;
+      const hasExternalUrl = hrefUrl !== blockSelectedUrl;
```

Comment:
If I'm understanding this correctly it seems this is trying to find the block associated with the selected link. This was not working properly for me when selecting the entirety of a link's text, or when clicking inside a link after making the changes I suggested to `hooks.js`

Reply:
Yes, this is trying to validate if there is a external link or has jump to link selected, but in a logic below it checks if has a parent node and I was not having account that so that was the problem